packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -22,34 +23,62 @@ import {
   Unsubscribe,
 } from '@coveo/headless';
 import {InitializeEvent} from '../../utils/initialization-utils';
+import i18next, {i18n} from 'i18next';
+import Backend, {BackendOptions} from 'i18next-http-backend';
+import LanguageDetector, {
+  DetectorOptions,
+} from 'i18next-browser-languagedetector';
 
 @Component({
   tag: 'atomic-search-interface',
   shadow: true,
+  assetsDirs: ['assets'],
 })
 export class AtomicSearchInterface {
   @Element() host!: HTMLDivElement;
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'info';
+  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
-  @State() engine?: Engine;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
   private initialized = false;
+  private afterInitializationCallbacks: (() => void)[] = [];
 
   componentWillLoad() {
     if (this.sample) {
       this.initialize(HeadlessEngine.getSampleConfiguration());
     }
+
+    this.i18n
+      .use(LanguageDetector)
+      .use(Backend)
```

Comment:
What is `Backend` for ?

Reply:
fetch the json files from the assets

Reply:
This is nice, fetching the JSON files should help minimize the payload.

One concern though, is whether there is/will be a flash where the language keys appear untranslated while we are retrieving the JSON? If so, do the docs mention anything around this?

Reply:
I'll rework the initialization mechanism a bit, it's far from perfect. I'll also add the "loading state", we should display the full components until strings are loaded.